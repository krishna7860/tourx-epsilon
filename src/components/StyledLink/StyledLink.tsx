import React from "react";
import Link from "./style";

interface LinkProps {
  to: string;
  children: JSX.Element;
}

const StyledLink = (props: LinkProps): JSX.Element => {
  const { to, children } = props;
  return <Link to={to}>{children}</Link>;
};

export default StyledLink;
