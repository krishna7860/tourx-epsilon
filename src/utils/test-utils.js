/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from "react";
import { render as rtlRender } from "@testing-library/react";
// import { renderHook as rtlRenderHooks } from "@testing-library/react-hooks";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import * as themes from "../theme/schema.json";
import reducer from "../redux/rootReducer";
// import defaultStore from "../redux/defaultStore";

function render(ui, options = {}) {
  const {
    initialState = {},
    store = createStore(reducer, initialState),
    theme,
    ...renderOptions
  } = options;

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <ThemeProvider theme={themes.data.light}>{children}</ThemeProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// function renderHookWithRedux(callback, options = {}) {
//   const { initialState = defaultStore, store = createStore(reducer, initialState), ...others } = options

//   const returned = rtlRenderHooks(callback, {
//     ...others,
//     // eslint-disable-next-line react/display-name
//     wrapper: ({ children }) => <Provider store={store}>{children}</Provider>
//   })

//   return {
//     ...returned,
//     store
//   }
// }

// re-export everything
export * from "@testing-library/react";
// override render method
export { render, rtlRender as renderHook };
