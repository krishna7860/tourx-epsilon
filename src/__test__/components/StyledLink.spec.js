/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { MemoryRouter } from "react-router";
import { render } from "../../utils/test-utils";
import StyledLink from "../../components/StyledLink/StyledLink";

describe("components/StyledLink", () => {
  it("should be rendered properly", () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={[`/`]}>
        <StyledLink to="/">Sample Link</StyledLink>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
