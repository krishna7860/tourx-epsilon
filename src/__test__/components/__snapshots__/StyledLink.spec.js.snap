// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/StyledLink should be rendered properly 1`] = `
<DocumentFragment>
  <a
    class="sc-bdnxRM fTfPUG"
    href="/"
  >
    Sample Link
  </a>
</DocumentFragment>
`;
